basic template for hpge post processing

tdixon97 opened the pull request:
Draft PR that we can add to with the basic hpge post processing
- [x] utils for IO
- [ ] basic windowing from evtid
- [ ] windowing from times
- [ ] detector response model (resolution)
- [ ] evt tier
- [ ] dead layers etc

ManuelHu commented:
Please also add a (possibly empty) `__init__.py` to the hpge directory. Otherwise the automatic API docs are not built... (I also just found that out in #3)

ManuelHu commented on src/reboost/hpge/processors.py:
```diff
@@ -0,0 +1,77 @@
+
+import awkward as ak
+import numpy as np
+import utils
```
```suggestion
from . import utils
```

ManuelHu commented on src/reboost/hpge/hit.py:
```diff
@@ -0,0 +1,39 @@
+
+import logging
+
+from reboost.hpge import processors, utils
```
```suggestion
from . import processors, utils
```

ManuelHu commented:
do not worry that all CI fails, this is a known problem and currently affects all our packages that (transitively) depend on pint for units. see https://github.com/hgrecco/pint/issues/1969

gipert commented on src/reboost/hpge/cli.py:
```diff
@@ -0,0 +1,105 @@
+        from reboost.hpge.hit import build_hit
+
+        with Path.open(Path(args.pars)) as config_f:
+            pars = yaml.safe_load(config_f)
```
YAML here but the configs in the docstrings are in JSON? Could you update the docstrings? Also, you can copy and use this from pygama in utils: https://github.com/legend-exp/pygama/blob/main/src/pygama/utils.py#L87-L110

tdixon97 commented:
The tests will not pass until https://github.com/legend-exp/legend-pygeom-hpges/pull/51 is merged.

ManuelHu commented on src/reboost/hpge/utils.py:
```diff
@@ -31,6 +31,14 @@
         "last_global_evtid",
     ],
 )
+FileInfo.__doc__ = "NamedTuple storing the information on the input files"
```
there should be a nicer way to have namedtuples with docs (i.e. by subclassing)

example: https://github.com/legend-exp/legend-pygeom-l200/blob/8ab9cdaae3a1d8ab0986beaebc704a5e91711795/src/l200geom/core.py#L17-L37

ManuelHu commented:
and to make the test on macOS pass more, you have to add something like
https://github.com/legend-exp/legend-pygeom-l200/blob/main/.github/workflows/main.yml#L34-L37
to the workflow. `pyg4ometry` does not contain all its dependencies in the macOS wheels.

tdixon97 commented:
Updated to be a bit more flexible, having a user defined "local" dictionary to `eval`. I also added more detailed docs (can check to understand the code logic). What do you think @gipert @ManuelHu ?

ManuelHu commented on src/reboost/hpge/hit.py:
```diff
@@ -0,0 +1,463 @@
+                        "expression": "reboost.hpge.processors.group_by_time(stp,window=10)"
+                    },
+                    "locals": {
+                        "hpge": "reboost.hpge.utils(meta_path=meta,pars=pars,detector=detector)"
```
```suggestion
                        "hpge": "reboost.hpge.utils.get_hpge(meta_path=meta,pars=pars,detector=detector)"
```
isn't there a function name missing?

tdixon97 replied:
yeah it was a mistake, i will add

gipert commented:
docs still need a lot of work and fixes but we can improve them in a future PR. you should rewrite the git history to remove the pre-commit bot commits before merging, I don't want to merge that junk. You just need to `git rebase -i <hash of your first commit>`, follow instructions to squash pre-commit bot commits into the previous, then force push

tdixon97 commented:
I think its worth taking time to cleanup and review things now. We could then review together in Munich.
________________________________
From: Luigi Pertoldi ***@***.***>
Sent: Tuesday, November 19, 2024 11:47 PM
To: legend-exp/reboost ***@***.***>
Cc: Dixon, Toby ***@***.***>; Author ***@***.***>
Subject: Re: [legend-exp/reboost] basic template for hpge post processing (PR #2)


⚠ Caution: External sender


docs still need a lot of work and fixes but we can improve them in a future PR. you should rewrite the git history to remove the pre-commit bot commits before merging, I don't want to merge that junk. You just need to git rebase -i <hash of your first commit>, follow instructions to squash pre-commit bot commits into the previous, then force push

—
Reply to this email directly, view it on GitHub<https://github.com/legend-exp/reboost/pull/2#issuecomment-2486905744>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANSET43WLXD5OQTJABMTBRL2BO5XLAVCNFSM6AAAAABQK3HXK2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBWHEYDKNZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

gipert commented:
Yes, let's keep this open and merge it together.

ManuelHu commented on .pre-commit-config.yaml:
```diff
@@ -57,6 +57,17 @@ repos:
         additional_dependencies:
           - pytest
 
+  - repo: https://github.com/kynan/nbstripout
+    rev: "0.7.1"
+    hooks:
+      - id: nbstripout
+        args:
+          [
+            "--drop-empty-cells",
+            "--extra-keys",
+            "metadata.kernelspec metadata.language_info",
+          ]
+
```
```suggestion
```

please remove again...?

ManuelHu replied:
We can also keep it, but we don't have notebooks in the repo any more?

tdixon97 commented:
For now I close this, will make a new PR
